pkg/kubelet/kubelet_pods.go

```diff
@@ -326,7 +326,19 @@ func ensureHostsFile(fileName, hostIP, hostName, hostDomainName string, hostAlia
 		hostsFileContent = managedHostsFileContent(hostIP, hostName, hostDomainName, hostAliases)
 	}
 
-	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	err = ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	if err != nil {
+		return err
+	}
+	fileInfo, err := os.Stat(fileName)
+	if err != nil {
+		return err
+	}
+	filePerms := fileInfo.Mode().Perm()
+	if filePerms != 0644 {
```

Comment:
note: `ioutil.WriteFile` doesn't set the permissions if the file exists, but I don't think it matters in this case.